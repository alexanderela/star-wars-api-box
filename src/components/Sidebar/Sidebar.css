@import url("https://fonts.googleapis.com/css?family=Pathway+Gothic+One")

.Sidebar {
  height: 20rem;
  overflow: hidden;
  width: 100%;
}

.Sidebar-hidden {
  display: none;
}

.fade {
  background-image: linear-gradient(0deg, transparent, black 75%);
  min-height: 20vh;
  position: relative;
  top: -25px;
  width: 100%;
  z-index: 1;
}

.star-wars {
  background: url('../../assets/SWAPI-background.jpg');
  color: #feda4a;
  display: flex;
  font-family: 'Pathway Gothic One', sans-serif;
  font-size: 550%;
  font-weight: 600;
  height: 30rem;
  justify-content: center;
  letter-spacing: 4px;
  line-height: 125%;
  overflow: hidden;
  perspective: 400px;
  position: relative;
  text-align: justify;
  top: -1rem;
  word-spacing: normal;
  z-index: 1;
}

.crawl {
  animation: crawl 90s linear;
  position: relative;
  transform-origin: 50% 100%;
}

.title {
  text-align: center;
}

.film-title {
  font-size: 150%;
  font-weight: 600;
}

@keyframes crawl {
  0% {
    top: 0px;
    transform: rotateX(20deg)  translateZ(0);
  }
  100% { 
    top: -6000px;
    transform: rotateX(25deg) translateZ(-2500px);
  }
}
