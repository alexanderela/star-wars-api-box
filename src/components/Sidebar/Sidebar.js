import React, { Component } from 'react';
import './Sidebar.css';
import PropTypes from 'prop-types';

class Sidebar extends Component {
	constructor(props) {
		super(props);
		this.state = {	
		}
	}

	romanizeNum = (episodeNum) => {
	  let roman = ''
	  let romanNums = [
	  	'I','II','III','IV','V','VI','VII','VIII','IX', 'X'
	  ]
	  romanNums.forEach(num => {
	  	if(romanNums[episodeNum]) {
	  		roman = romanNums[episodeNum]
	  	}
	  })
	  return roman
	}

	render() {
		const { films } = this.props
		return (
			<div className="Sidebar fade">
				<div className="star-wars">
					<div className="crawl">	    
			    <div className="title">
			      <p>Episode {this.romanizeNum(films.episode_id)}</p>
			      <h1 className="film-title">{films.title}</h1>
			      <p>Released:  {films.date}</p>
			    </div>
			    <p>{films.opening_crawl}</p>
			  	</div>
				</div>
			</div>
		);
	}
}

Sidebar.propTypes = {
	films: PropTypes.object
}

export default Sidebar;