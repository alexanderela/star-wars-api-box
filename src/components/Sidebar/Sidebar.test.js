/* eslint-disable */
import React from 'react';
import ReactDOM from 'react-dom';
import Sidebar from './Sidebar';
import { shallow } from 'enzyme';

describe('Sidebar', () => {
	let wrapper;

	beforeEach(() => {
		const mockFilm = {"opening_crawl": "Oh hayyyyy...", "episode_id": 3, 
		"title": "Revenge of the Sith"}
		wrapper = shallow(<Sidebar films={mockFilm}/>);

	})
	
	it('matches the snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	})

})