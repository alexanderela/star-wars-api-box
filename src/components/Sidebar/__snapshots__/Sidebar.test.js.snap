// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Sidebar
    films={
      Object {
        "episode_id": 3,
        "opening_crawl": "Oh hayyyyy...",
        "title": "Revenge of the Sith",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="star-wars"
      >
        <div
          className="crawl"
        >
          <div
            className="title"
          >
            <p>
              Episode 
              IV
            </p>
            <h1
              className="film-title"
            >
              Revenge of the Sith
            </h1>
            <p>
              Released:  
            </p>
          </div>
          <p>
            Oh hayyyyy...
          </p>
        </div>
      </div>,
      "className": "Sidebar fade",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="crawl"
        >
          <div
            className="title"
          >
            <p>
              Episode 
              IV
            </p>
            <h1
              className="film-title"
            >
              Revenge of the Sith
            </h1>
            <p>
              Released:  
            </p>
          </div>
          <p>
            Oh hayyyyy...
          </p>
        </div>,
        "className": "star-wars",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="title"
            >
              <p>
                Episode 
                IV
              </p>
              <h1
                className="film-title"
              >
                Revenge of the Sith
              </h1>
              <p>
                Released:  
              </p>
            </div>,
            <p>
              Oh hayyyyy...
            </p>,
          ],
          "className": "crawl",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  Episode 
                  IV
                </p>,
                <h1
                  className="film-title"
                >
                  Revenge of the Sith
                </h1>,
                <p>
                  Released:  
                </p>,
              ],
              "className": "title",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Episode ",
                    "IV",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Episode ",
                  "IV",
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Revenge of the Sith",
                  "className": "film-title",
                },
                "ref": null,
                "rendered": "Revenge of the Sith",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Released:  ",
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Released:  ",
                  undefined,
                ],
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Oh hayyyyy...",
            },
            "ref": null,
            "rendered": "Oh hayyyyy...",
            "type": "p",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="star-wars"
        >
          <div
            className="crawl"
          >
            <div
              className="title"
            >
              <p>
                Episode 
                IV
              </p>
              <h1
                className="film-title"
              >
                Revenge of the Sith
              </h1>
              <p>
                Released:  
              </p>
            </div>
            <p>
              Oh hayyyyy...
            </p>
          </div>
        </div>,
        "className": "Sidebar fade",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="crawl"
          >
            <div
              className="title"
            >
              <p>
                Episode 
                IV
              </p>
              <h1
                className="film-title"
              >
                Revenge of the Sith
              </h1>
              <p>
                Released:  
              </p>
            </div>
            <p>
              Oh hayyyyy...
            </p>
          </div>,
          "className": "star-wars",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="title"
              >
                <p>
                  Episode 
                  IV
                </p>
                <h1
                  className="film-title"
                >
                  Revenge of the Sith
                </h1>
                <p>
                  Released:  
                </p>
              </div>,
              <p>
                Oh hayyyyy...
              </p>,
            ],
            "className": "crawl",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    Episode 
                    IV
                  </p>,
                  <h1
                    className="film-title"
                  >
                    Revenge of the Sith
                  </h1>,
                  <p>
                    Released:  
                  </p>,
                ],
                "className": "title",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Episode ",
                      "IV",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Episode ",
                    "IV",
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Revenge of the Sith",
                    "className": "film-title",
                  },
                  "ref": null,
                  "rendered": "Revenge of the Sith",
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Released:  ",
                      undefined,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Released:  ",
                    undefined,
                  ],
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Oh hayyyyy...",
              },
              "ref": null,
              "rendered": "Oh hayyyyy...",
              "type": "p",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
